test/app/App/Panel/AddToken/index.test.js

```diff
@@ -62,50 +79,60 @@ describe('selecting token chain', () => {
 
 describe('retrieving token metadata', () => {
   it('should display successfully loaded data', async () => {
-    const { findByLabelText, getByText, getByLabelText, getByRole } = render(
+    const { getByText, getByLabelText, getByRole, rerender } = render(
       <AddToken activeChains={[
         { id: 1, name: 'Mainnet', connection: { primary: { connected: true } } },
         { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }
       ]} />
     )
-
-    await user.click(getByText('Polygon'))
-
-    const contractAddressInput = await findByLabelText(`What is the token's contract address?`)
-    await user.type(contractAddressInput, '0x3432b6a60d23ca0dfca7761b7ab56459d9c964d0')
-    const setAddressButton = getByRole('button')
-    await user.click(setAddressButton)
-
+    rerenderComponent = rerender
+
+    const selectPolygonButton = getByText('Polygon')
+    await user.click(selectPolygonButton)
+    
+    await waitFor(async () => {
```

Comment:
don't think we need to use `waitFor` here. as I understand it, this allows you to wait "indefinitely" for something to render if it depends on a promise since it's hard to get promises to resolve when you want them to in unit tests. wouldn't this work just the same without this?

Reply:
No, it's configurable and as I understand the docs, waits for 1000ms by default

https://testing-library.com/docs/dom-testing-library/api-async/#waitfor

I know you aren't a fan of it though so yes, I have tried without.  Doesn't work.

Reply:
Feel free to have a play but I think these tests are ok as is...the re-render could be abstracted as we'll need this approach for anything using the nav but it's not something I want to spend time on right now.